mlflow/environment_variables.py

```diff
@@ -186,3 +186,7 @@ def get(self):
 MLFLOW_SCORING_SERVER_REQUEST_TIMEOUT = _EnvironmentVariable(
     "MLFLOW_SCORING_SERVER_REQUEST_TIMEOUT", int, 60
 )
+
+#: Specifies whether or not to use the stdin scoring server when making predictions
+#: in `mlflow.evaluate` and `mlflow.pyfunc.spark_udf` (default: ``False``).
+MLFLOW_USE_STDIN_SERVER = _BooleanEnvironmentVariable("MLFLOW_USE_STDIN_SERVER", False)
```

Comment:
We might not need this environment variable. Instead, we can use a spark conf that is set to True in a shared cluster.

Reply:
Spark conf sounds better :)

Reply:
We should make sure it starts with `spark.databricks.`

Reply:
I found we already have `spark.databricks.unityCatalog.enabled`. Can we use it?

```
if spark.conf.get("spark.databricks.unityCatalog.enabled") == 'true':
    # Use the stdin server
else:
    ... 
```

Reply:
does `spark.databricks.unityCatalog.enabled` mean we're on a shared cluster or just that UC is enabled? We should only use the file-based server if we're on a shared cluster.

Reply:
We decided to directly check the port connectivity :)